2017/proposals/bansal-utkarsh.md

```diff
@@ -0,0 +1,628 @@
+pytest does not collect classes with an `__init__` method (which I've explained above)
+
+Also to get an approximate of how much time and effort _part 2_ would take, I have worked on a part of issue [#516](https://github.com/MDAnalysis/mdanalysis/issues/516)
+where I fixed the `GRO` writers API to follow the reader API standard and modified test classes to inherit from base test classes. As of now the changes have been
+approved and are ready to be merged. See [#1196](https://github.com/MDAnalysis/mdanalysis/pull/1196)
+
+
+Schedule of Deliverables
+========================
+
+### May 4th - May 29th, **Community Bonding Period**
+
+### May 30th - June 3rd
+* Begin work on **Part 1**. Start by editing tests to make them discoverable by pytest.
+Also actively discuss the `__init__` issue as it will be dealt with on a case by case
+basis.
+* Send a PR only for review and discussion purposes. It will _not_ be ready to merge.
+
+### June 5th - June 9th
+* Complete the test discovery issue and get _atleast_ one plugin working with pytest.
+* Work is done on the same PR. Still _not_ ready to merge.
+* Get a _bare minimum_ TravisCI setup to work.
+
+### June 12th - June 16th
+* Get all the plugins to work
+* Work is done on the same PR. Still _not_ ready to merge.
+
+### June 19th - June 23th, **End of Phase 1**
+* Complete configuring TravisCI with plugins, code coverage and quantified code.
+* Final review, documentation and code cleanup as suggested.
+* Get the PR merged.
+
+### June 26 - June 30th, **Begin of Phase 2**
+* Begin working on **Part 2** - the coordinates module. PRs will we worked on in parallel as the
+coordinate tests aren't interdependent. So I can work on one while another one is waiting for a review.
+* Port 4 coordinate test files - 4 PRs
+
+### July 3rd - July 7th
+* Port another 6 coordinate test files - 6 PRs
+
+### July 10th - July 14th
+* Port another 6 coordinated test files - 6 PRs
+
+### July 17th - July 21th, **End of Phase 2**
+* Port another 6 coordinate test files - 6 PRs
+
+### July 24th - July 28th, **Begin of Phase 3**
+* Port remaining 4 coordinate test files - 4 PRs
+
+### July 31st - August 4th
+* Begin work on **Part 3**. PRs are again worked on in parallel.
+* Port analysis & auxiliary modules - multiple PRs
+
+### August 7th - August 11th
+* Port core & topology modules - multiple PRs
+
+### August 14th - August 18th
+* Port files in the root directory, lib and formats modules - multiple PRs
+
+### August 21st - August 25th, **Final Week**
+* This week is set aside as a buffer to negate any time lost/lag in work. Follow up on reviews and get PRs
+merged into develop.
+
+### August 28th - August 29th, **Submit final work**
+* Submit all the work.
+
+## Future works
+All the points mentioned here are _optional_ and may be done if I'm ahead of schedule
+
+* Benchmarking of tests module.
+* Add support for doctests.
```

Comment:
I didn't dig much into this. Would you wan't to include this in the current work too?  If yes, I'll have to read relevant docs before working on the timeline.